test/foo/lib/config/test_configloader.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import Mock
+
+import pytest
+
+from foo.lib.config import ConfigLoader, ConfigLoaderException
+
+
+def reset_config_loader_singleton():
+    ConfigLoader._instance = None
+    ConfigLoader._instance_ready = False
```

Comment:
Modify private attributes from class breaks the elementary OOP rule. Here I can ask the main question from this project, If we really need's to use the Singleton pattern ? I don't want to paste multiple articles about why he is almost always bad choice, antipattern etc, but let's thinking about do we really need it here in this project.